nixos/modules/services/web-apps/pairdrop.nix

```diff
@@ -0,0 +1,154 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+let
+  inherit (lib)
+    boolToString
+    getExe
+    isAttrs
+    isBool
+    maintainers
+    mapAttrs
+    mkEnableOption
+    mkIf
+    mkOption
+    mkPackageOption
+    optionalAttrs
+    types
+    ;
+
+  cfg = config.services.pairdrop;
+
+  json = pkgs.formats.json { };
+in
+{
+  options.services.pairdrop = {
+    enable = mkEnableOption "pairdrop";
+
+    package = mkPackageOption pkgs "pairdrop" { };
+
+    port = mkOption {
+      type = types.port;
+      default = 3000;
+      example = 3010;
+      description = "The port to listen on.";
+    };
+
+    settings = mkOption {
+      description = ''
+        Additional configuration (environment variables) for PairDrop, see
+        <https://github.com/schlagmichdoch/PairDrop/blob/master/docs/host-your-own.md#environment-variables>
+        for supported values.
+      '';
+
+      type = types.submodule {
+        freeformType =
+          with types;
+          attrsOf (oneOf [
+            bool
+            int
+            str
+          ]);
+
+        options = {
+          RTC_CONFIG = mkOption {
+            type = types.oneOf [
+              json.type
+              types.path
+            ];
+            default = null;
+            example = {
+              sdpSemantics = "unified-plan";
+              iceServers = [
+                {
+                  urls = "stun:stun.example.com:19302";
+                }
+              ];
+            };
+            description = ''
+              Configuration for STUN/TURN servers.
+              If this is an attribute set, it is converted to JSON and
+              written into a file automatically.
+              Otherwise, a file path is expected.
+            '';
+          };
+        };
+      };
+
+      default = { };
+
+      example = {
+        DEBUG_MODE = true;
+        RATE_LIMIT = 1;
+        IPV6_LOCALIZE = 4;
+        WS_FALLBACK = true;
+        SIGNALING_SERVER = "pairdrop.net";
+
+        DONATION_BUTTON_ACTIVE = false;
+        TWITTER_BUTTON_ACTIVE = false;
+        MASTODON_BUTTON_ACTIVE = false;
+        BLUESKY_BUTTON_ACTIVE = false;
+        CUSTOM_BUTTON_ACTIVE = false;
+        PRIVACYPOLICY_BUTTON_ACTIVE = false;
+      };
+    };
+  };
+
+  config = mkIf cfg.enable {
+    systemd.services.pairdrop =
+      let
+        rtcConfig = cfg.settings.RTC_CONFIG;
+        primitiveSettings = builtins.removeAttrs cfg.settings [ "RTC_CONFIG" ];
+        environment =
+          {
+            PORT = toString cfg.port;
+          }
+          // (mapAttrs (_: v: if isBool v then boolToString v else toString v) primitiveSettings)
```

Comment:
```suggestion
          // (mapAttrs lib.toString primitiveSettings)
```

Reply:
`lib.toString` does not exist :thinking:
```
nix-repl> :l <nixpkgs>
Added 24722 variables.

nix-repl> lib.toString true
error: attribute 'toString' missing
       at «string»:1:1:
            1| lib.toString true
             | ^
       Did you mean isString?

nix-repl> lib.boolToString true
"true"
```